modules/data-entry/src/main/java/io/uhndata/cards/dataentry/internal/serialize/QuestionnaireToCsvProcessor.java

```diff
@@ -0,0 +1,261 @@
+                label = nodeJson.getString("text");
+            }
+            columns.add(label);
+            csvData.put(nodeJson.getString("jcr:uuid"), new ArrayList<>());
+        }
+    }
+
+    private void processFormsToRows(JsonArray formsJson, Map<String, List<String>> csvData)
+    {
+        formsJson.stream()
+            .filter(value -> ValueType.OBJECT.equals(value.getValueType()))
+            .map(JsonValue::asJsonObject)
+            .forEach(form -> processForm(form, csvData));
+    }
+
+    private void processForm(final JsonObject form, Map<String, List<String>> csvData)
+    {
+        if (form.containsKey("subject")) {
+            JsonObject subject = form.getJsonObject("subject");
+            csvData.get("Patient ID").add(subject.getString("identifier"));
+        }
+        csvData.get("Created").add(form.getString("jcr:created"));
+        processFormSection(form, csvData);
+    }
+
+    /**
+     * Converts a JSON serialization of an answer section to CSV text.
+     *
+     * @param answerSectionJson a JSON serialization of an answer section
+     * @param csvData the string builder where the serialization must be appended
+     */
+    private void processFormSection(final JsonObject answerSectionJson, Map<String, List<String>> csvData)
+    {
+        answerSectionJson.values().stream()
+            .filter(value -> ValueType.OBJECT.equals(value.getValueType()))
+            .map(JsonValue::asJsonObject)
+            .filter(value -> value.containsKey("jcr:primaryType"))
+            .forEach(value -> processFormElement(value, csvData));
+        //if (section.hasProperty("recurrent") && section.getProperty("recurrent").getBoolean())
+    }
+
+    /**
+     * Process answer to fill in data aggregator.
+     *
+     * @param nodeJson a JSON serialization of an answer node
+     * @param csvData data aggregator
+     */
+    @SuppressWarnings("checkstyle:CyclomaticComplexity")
+    private void processFormElement(final JsonObject nodeJson, Map<String, List<String>> csvData)
+    {
+        final String nodeType = nodeJson.getString("jcr:primaryType");
+        if ("cards:AnswerSection".equals(nodeType) || "cards:Section".equals(nodeType)) {
+            processFormSection(nodeJson, csvData);
+        } else if (nodeType.startsWith("cards:") && nodeType.endsWith("Answer")) {
+
+            String uuid = nodeJson.getJsonObject("question").getString("jcr:uuid");
+            List<String> answerColumn = csvData.get(uuid);
+
+            final JsonValue value = nodeJson.get("value");
+            if (value == null) {
+                // Fallback to "displayedValue"
+                final JsonValue displayedValue = nodeJson.get("displayedValue");
+                if (displayedValue == null) {
+                    answerColumn.add("");
+                } else {
+                    answerColumn.add(((JsonObject) displayedValue).getString("displayedValue"));
+                }
```

Comment:
Let's ignore the `displayedValue` in the first iteration and just work with `value`.
```suggestion
                answerColumn.add("");
```